app/core/api/v1/views/user.py

```diff
@@ -17,3 +18,10 @@ class PersonRetrieveView(RetrieveAPIView):
 
     def get_object(self):
         return self.request.user.person
+
+class UserRetrieveView(RetrieveAPIView):
```

Comment:
You can use a single view for updating and retrieving by overriding the get and put methods.

Reply:
Person Retrieve view was created by you. Have changed the UserRetrieveView to UserRetrieveUpdateView